### PR TITLE
Kotin --> Wasm now supports services. 

### PR DESCRIPTION
**Pull request by alxrsngrtn**

### CODE REVIEW

File: src/wasm/kotlin/wasm/arcs/Particle.kt [hunk 1]
```diff
@@ -183,6 +191,45 @@ class Collection<T : Entity<T>> constructor(private val entityCtor: () -> T) : H
...
+    companion object {
+
+      fun decodeDictionary(str: String): Map<String, String> {
+        val decoder = StringDecoder(str)
+        val dict = mutableMapOf<String, String>()
+
+        var num = decoder.getInt(":")
+
+        // while(num-- > 0)
+        while(num --> 0){
```
Comment (mykmartin) [inline src/wasm/kotlin/wasm/arcs/Particle.kt]: Just use the usual spacing?

File: src/wasm/kotlin/wasm/arcs/Particle.kt [hunk 2]
```diff
@@ -183,6 +191,45 @@ class Collection<T : Entity<T>> constructor(private val entityCtor: () -> T) : H
...
+        return dict
+      }
+
+      fun decodeList(str: String): List<String> {
+        val decoder = StringDecoder(str)
+
+        val list = mutableListOf<String>()
+
+        var num = decoder.getInt(":")
+        // while(num-- > 0)
```
Comment (shaper) [inline src/wasm/kotlin/wasm/arcs/Particle.kt]: Here too

**Comment by mykmartin** [comment]
> I sure wish we had unit tests for these, to consider for near future PRs? For both Kotlin and C++. And some light integration test so that we know if KT Wasm changes break C++ or vice versa.

FYI I'm currently working on wasm build integration (at least for C++) to enable testing.

**Comment by alxrsngrtn** [comment]
I'm glad you're writing integration tests! I'll see if I can re-use for Kotlin-wasm, or else re-write them.

**Pull request merged**